### PR TITLE
Component hooks: return only state, render function to be imported separately

### PR DESCRIPTION
**Pull request by dzearing**
Addresses RFC proposal #15497

This change decouples the unnecessarily coupled render function and state hook:

Before:
```jsx
import { useButton } from '@fluentui/react';

const Button = React.forwardRef((props, ref) => {
  const {state, render} = useButton(props, ref);

  return render(state);
});
```

After: 
```jsx
import { useButton, renderButton } from '@fluentui/react';

const Button = React.forwardRef((props, ref) => {
  const state = useButton(props, ref);

  return renderButton(state);
});
```

### CODE REVIEW

**Review by behowell — APPROVED** [review]
👍

File: packages/react-button/src/components/Button/index.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ export * from './Button.types';
 export * from './useButtonClasses';
 export * from './useButton';
 export * from './useButtonState';
+export * from './renderButton';
```
Comment (khmakoto) [inline packages/react-button/src/components/Button/index.ts:5]: This and `ToggleButton` are the only index files from which you are exporting a render function, is that on purpose?

**Review by khmakoto — APPROVED** [review]

File: packages/react-button/src/components/Button/index.ts (hunk 1 above)
Comment (dzearing) [reply packages/react-button/src/components/Button/index.ts:5]: no! Thank you! I will make sure they're all exported.

Comment (dzearing) [reply packages/react-button/src/components/Button/index.ts]: done.

**Pull request merged**